homeassistant/components/binary_sensor/tod.py

```diff
@@ -0,0 +1,221 @@
+"""
+Support for showing the binary sensors represending current time of the day.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.tod/
+
+"""
+from datetime import datetime, timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.binary_sensor import (
+    ENTITY_ID_FORMAT, PLATFORM_SCHEMA)
+from homeassistant.const import (
+    ATTR_ENTITY_ID, ATTR_FRIENDLY_NAME, CONF_AFTER, CONF_BEFORE, CONF_SENSORS,
+    STATE_OFF, STATE_ON, SUN_EVENT_SUNRISE, SUN_EVENT_SUNSET)
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity, async_generate_entity_id
+from homeassistant.helpers.event import async_track_point_in_time
+from homeassistant.helpers.sun import (
+    get_astral_event_date, get_astral_event_next)
+from homeassistant.util import dt as dt_util
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_AFTER_OFFSET = 'after_offset'
+CONF_BEFORE_OFFSET = 'before_offset'
+
+ATTR_NEXT_UPDATE = 'next_update'
+
+
+SENSOR_SCHEMA = vol.Schema({
+    vol.Required(CONF_AFTER): vol.Any(cv.time, vol.All(
+        vol.Lower, vol.Any(cv.sun_event))),
+    vol.Optional(CONF_AFTER_OFFSET, default=timedelta(0)): cv.time_period,
+    vol.Required(CONF_BEFORE): vol.Any(cv.time, vol.All(
+        vol.Lower, vol.Any(cv.sun_event))),
+    vol.Optional(CONF_BEFORE_OFFSET, default=timedelta(0)): cv.time_period,
+    vol.Optional(ATTR_FRIENDLY_NAME): cv.string,
+    vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_SENSORS): vol.Schema({cv.slug: SENSOR_SCHEMA}),
+})
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the ToD sensors."""
+    if hass.config.time_zone is None:
+        _LOGGER.error("Timezone is not set in Home Assistant configuration")
+        return False
+
+    sensors = []
+    for device, device_config in config[CONF_SENSORS].items():
+        after = device_config[CONF_AFTER]
+        after_offset = device_config[CONF_AFTER_OFFSET]
+        before = device_config[CONF_BEFORE]
+        before_offset = device_config[CONF_BEFORE_OFFSET]
+        sensors.append(
+            TodSensor(
+                hass, device, after, after_offset, before, before_offset
+            )
+        )
+
+    if not sensors:
+        _LOGGER.error("No sensors added")
+        return
+
+    async_add_entities(sensors)
+    return
```

Comment:
Unnecessary return

Reply:
OK, but explicit is better than implicit ;-)